resolver/addressable_resolver.go

```diff
@@ -56,6 +58,28 @@ func NewURIResolver(ctx context.Context, callback func(types.NamespacedName)) *U
 		},
 	}
 
+	ret.listerFactory = func(gvr schema.GroupVersionResource) (cache.GenericLister, error) {
+		_, l, err := ret.informerFactory.Get(ctx, gvr)
+		return l, err
+	}
+
+	return ret
+}
+
+// NewURIResolver constructs a new URIResolver with context and a callback
+// for a given listableType (Listable) passed to the URIResolver's tracker.
+// extra when controller context is used
+func NewURIResolverWithControllerContext(ctx context.Context, dmInformerFactory pkgapisduck.InformerFactory,
+	dmTracker tracker.Interface) *URIResolver {
+	ret := &URIResolver{}
+
+	ret.tracker = dmTracker
+
+	ret.listerFactory = func(gvr schema.GroupVersionResource) (cache.GenericLister, error) {
+		_, l, err := dmInformerFactory.Get(ctx, gvr)
+		return l, err
+	}
+
 	return ret
```

Comment:
we can collapse this:

```suggestion
	return &URIResolver{
	  tracker: dmTracker,
	  listerFactory = func(gvr schema.GroupVersionResource) (cache.GenericLister, error) {
		_, l, err := dmInformerFactory.Get(ctx, gvr)
		return l, err
	  },
	}
```